### PR TITLE
Podman Improvements

### PR DESCRIPTION
**Pull request by clebergnu**
This is a collection of small improvements and one fix to a race condition, where the state of the container (and thus the task) would not take into account the transition from "created" to "running".

### CODE REVIEW

File: avocado/plugins/spawners/podman.py [hunk 1]
```diff
@@ -110,6 +111,37 @@ class PodmanSpawner(DeploymentSpawner, SpawnerMixin):
     def __init__(self, config=None, job=None):  # pylint: disable=W0231
         SpawnerMixin.__init__(self, config, job)
         self.environment = f"podman:{self.config.get('spawner.podman.image')}"
+        self._podman_version = (None, None, None)
+        self._podman = None
```
Comment (richtja) [inline avocado/plugins/spawners/podman.py:116]: IIUIC this is not related to this commit, am I right?

File: avocado/plugins/spawners/podman.py [hunk 2]
```diff
@@ -142,6 +142,17 @@ def podman_version(self):
         if self._podman_version == (None, None, None):
             self._podman_version = self._get_podman_version()
         return self._podman_version
+        self._podman = None
```
Comment (richtja) [inline avocado/plugins/spawners/podman.py]: This will never be used, I think it is a mistake.

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @clebergnu, thank you for this update. Overall the changes LGTM, there is just one small issue with `_podman` variable, I think it should be part of different commit. Can you please update it? Thank you.

File: avocado/plugins/spawners/podman.py [hunk 3]
```diff
@@ -142,7 +143,6 @@ def podman_version(self):
         if self._podman_version == (None, None, None):
             self._podman_version = self._get_podman_version()
         return self._podman_version
-        self._podman = None
```
Comment (richtja) [inline avocado/plugins/spawners/podman.py]: I can see that you removed it here. Can you please add this to the proper commit? Thank you.

**Comment by clebergnu** [comment]
> Hi @clebergnu, thank you for this update. Overall the changes LGTM, there is just one small issue with `_podman` variable, I think it should be part of different commit. Can you please update it? Thank you.

Good catch!  It was a bad rebase on my side.

I've pushed an updated version.

**Review by richtja — APPROVED** [review]
LGTM
